### PR TITLE
Add vscode GitHub Pull Requests plugin

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
Add vscode GitHub Pull Requests plugin to the plugins list page.

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/CRW-1226

### CODE REVIEW

File: dependencies/che-plugin-registry/v3/plugins/ms-vscode/vscode-github-pullrequest/0.20.0/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+name: vscode-github-pullrequest
+version: 0.20.0
+type: VS Code extension
+displayName: GitHub Pull Requests and issues
+title: Pull Request Provider for GitHub.
+description: This extension allows you to review and manage GitHub pull requests and issues.
+icon: https://raw.githubusercontent.com/microsoft/vscode-pull-request-github/0.20.0/resources/icons/github_logo.png
+repository: https://github.com/microsoft/vscode-pull-request-github
+category: Other
+firstPublicationDate: '2020-10-8'
```
Comment (nickboldt) [inline dependencies/che-plugin-registry/v3/plugins/ms-vscode/vscode-github-pullrequest/0.20.0/meta.yaml:12]: date format is incorrect; please also update to this week's date.

File: dependencies/che-plugin-registry/v3/plugins/ms-vscode/vscode-github-pullrequest/0.20.0/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+publisher: ms-vscode
+apiVersion: v2
+name: vscode-github-pullrequest
+version: 0.20.0
+type: VS Code extension
+displayName: GitHub Pull Requests and issues
+title: Pull Request Provider for GitHub.
+description: This extension allows you to review and manage GitHub pull requests and issues.
+icon: https://raw.githubusercontent.com/microsoft/vscode-pull-request-github/0.20.0/resources/icons/github_logo.png
```
Comment (nickboldt) [inline dependencies/che-plugin-registry/v3/plugins/ms-vscode/vscode-github-pullrequest/0.20.0/meta.yaml:9]: use consistent icon in CRW https://github.com/redhat-developer/codeready-workspaces/blob/crw-2.6-rhel-8/dependencies/che-plugin-registry/v3/plugins/redhat/java11/0.69.0/meta.yaml#L9

```
icon: https://raw.githubusercontent.com/redhat-developer/codeready-workspaces/master/dependencies/che-plugin-registry/resources/images/default.svg?sanitize=true
```

File: dependencies/che-plugin-registry/v3/plugins/ms-vscode/vscode-github-pullrequest/0.20.0/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+repository: https://github.com/microsoft/vscode-pull-request-github
+category: Other
+firstPublicationDate: '2020-10-8'
+spec:
+  extensions:
+  - https://github.com/microsoft/vscode-pull-request-github/releases/download/0.20.0/vscode-pull-request-github-0.20.0.vsix
```
Comment (nickboldt) [inline dependencies/che-plugin-registry/v3/plugins/ms-vscode/vscode-github-pullrequest/0.20.0/meta.yaml:15]: latest release is 0.21, please update.

**Review by nickboldt — CHANGES_REQUESTED** [review]
-1, changes requested inline

File: dependencies/che-plugin-registry/v3/plugins/ms-vscode/vscode-github-pullrequest/0.20.0/meta.yaml (hunk 3 above)
Comment (nickboldt) [reply dependencies/che-plugin-registry/v3/plugins/ms-vscode/vscode-github-pullrequest/0.20.0/meta.yaml:15]: also unless we control the build process for creating this .vsix file, we should not reference binaries on GH directly, but should use the version we've rebuilt and published on https://download.jboss.org/jbosstools/

**Comment by nickboldt** [comment]
Assuming that this new plugin will need to be documented as it'll improve git-related workflows in Che/CRW, please also link to 

```
Docs PR (if applicable)
```

or open a related RHDEVDOCS issue to track the work separately.

**Comment by ericwill** [comment]
Building this extension on the jenkins pipeline fails (of course). This PR may have to wait until I get the Dockerfile based build system up and running (hopefully before Christmas).

**Comment by MichalMaler** [comment]
@ericwill Hello Eric! Let me know when I can put this back in CRW docs. I think it will be in January, right? Cheers

**Comment by ericwill** [comment]
> @ericwill Hello Eric! Let me know when I can put this back in CRW docs. I think it will be in January, right? Cheers

Yes, early January sounds good.

**Comment by nickboldt** [comment]
@vinokurig can you apply requested changes to this PR so we can merge it into 2.6?

**Comment by ericwill** [comment]
We still need to build the GitHub PR plugin ourselves, in https://github.com/redhat-developer/codeready-workspaces-vscode-extensions

I started this work but it's still pending.

**Comment by vinokurig** [comment]
I managed to build the plugin in my fork: https://github.com/vinokurig/codeready-workspaces-vscode-extensions/releases/tag/v7076c43 by updating the plugin version to 0.21.4. The build of current 0.20.0 version fails, probably the author did some patches to publish it. The problem is that the 0.21.4 version doesn't work in che-theia due to missing API. I've filed an issue to update the plugin in Che: https://github.com/eclipse/che/issues/18661

**Comment by ericwill** [comment]
> I managed to build the plugin in my fork: https://github.com/vinokurig/codeready-workspaces-vscode-extensions/releases/tag/v7076c43 by updating the plugin version to 0.21.4. The build of the current 0.20.0 version fails, probably the author did some patches to publish it. The problem is that the 0.21.4 version doesn't work in che-theia due to missing API. I've filed an issue to update the plugin in Che: [eclipse/che#18661](https://github.com/eclipse/che/issues/18661)

Florent was able to build the extension, here is the link: https://github.com/redhat-developer/codeready-workspaces-vscode-extensions/releases/download/vdb8b64f/vscode-pull-request-github-0.20.0.vsix

Let's include 0.20.0 in CRW 2.6 and we can explore updating to 0.21.4 later on.

File: dependencies/che-plugin-registry/v3/plugins/ms-vscode/vscode-github-pullrequest/0.20.0/meta.yaml (hunk 3 above)
Comment (ericwill) [reply dependencies/che-plugin-registry/v3/plugins/ms-vscode/vscode-github-pullrequest/0.20.0/meta.yaml:15]: The latest [needs some missing theia API](https://github.com/redhat-developer/codeready-workspaces/pull/363#issuecomment-748057471) so we won't be upgrading directly to that. We can however include version 0.20.0 of the plugin in CRW 2.6, as the extension has been built by us now.

**Comment by MichalMaler** [comment]
Great stuff and nice progress! Let me know when I can return the docs about this plugin back to CRW. BTW, do we need to update the Che-relative content?

https://www.eclipse.org/che/docs/che-7/end-user-guide/version-control/#accessing-a-git-repository-via-ssh_version-control

**Comment by ericwill** [comment]
> Great stuff and nice progress! Let me know when I can return the docs about this plugin back to CRW. BTW, do we need to update the Che-relative content?
> 
> https://www.eclipse.org/che/docs/che-7/end-user-guide/version-control/#accessing-a-git-repository-via-ssh_version-control

When this PR is merged then you can restore the docs. As for the content, I'll have Igor verify the content with you when he gets back from PTO next week.

**Comment by MichalMaler** [comment]
Ok, great. Thx Eric

**Comment by ericwill** [comment]
> OK, I'm good to merge this once someone fixes the date `'2020-10-8'` ... not only is the day only a single digit, it's 3 months ago. Please bump that to a date in Jan 2021 for accuracy.

I've pushed a new version to fix the date and the vsix reference, I think we can merge this one now

**Comment by vinokurig** [comment]
I've opened a doc's PR that enables the GitHub documentation for CRW: https://github.com/eclipse/che-docs/pull/1784

**Comment by vinokurig** [comment]
@nickboldt Could you please review it again?

**Review by nickboldt — APPROVED** [review]

**Pull request merged**